addons/binding/org.openhab.binding.regoheatpump/src/main/java/org/openhab/binding/regoheatpump/handler/HusdataHandler.java

```diff
@@ -0,0 +1,262 @@
+    public void dispose() {
+        super.dispose();
+
+        scheduledRefreshFuture.cancel(true);
+        scheduledRefreshFuture = null;
+
+        connection.close();
+        connection = null;
+
+        mappings = null;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    private synchronized void handleDataFromHusdataInterface() {
+
+        final RegoConnection connection = this.connection;
+
+        while (Thread.interrupted() == false) {
+
+            try {
+
+                if (connection.isConnected() == false) {
+
+                    bufferedReader = null;
+                    connection.connect();
+
+                    // Request real-time registers.
+                    logger.debug("Requesting read and dump of real-time registers.");
+                    final OutputStream outputStream = connection.outputStream();
+                    outputStream.write(new String("XR\r\n").getBytes());
+                    outputStream.flush();
+                }
+
+                if (bufferedReader == null) {
+                    bufferedReader = new BufferedReader(new InputStreamReader(connection.inputStream()));
+                }
+
+                final String line = bufferedReader.readLine();
+                if (line == null) {
+                    throw new EOFException();
+                }
+
+                if (line.isEmpty()) {
+                    continue;
+                }
+
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Got '{}'", line);
+                }
+
+                processReceivedData(line);
+
+            } catch (SocketTimeoutException e) {
+
+                // Do nothing. Just happen to allow the thread to check if it has to stop.
+                break;
+
+            } catch (IOException e) {
+
+                logger.warn("Processing request failed", e);
+
+                bufferedReader = null;
+
+                if (Thread.interrupted() == false) {
+                    connection.close();
+                    updateStatus(ThingStatus.OFFLINE);
+                }
+
+                break;
+
+            } catch (Exception e) {
+
+                logger.warn("Error occurred during message waiting", e);
+                break;
+            }
+        }
+
+        // If state here is still unknown, than something went wrong so set thing status to OFFLINE.
+        if (thing.getStatus() == ThingStatus.UNKNOWN) {
+            updateStatus(ThingStatus.OFFLINE);
+        }
+    }
+
+    private void processReceivedData(final String line) {
+        if (line.length() != 10) {
+            logger.debug("Unexpected length for '{}'", line);
+            return;
+        }
+
+        if (line.charAt(0) != 'X' || line.charAt(1) != 'R') {
+            logger.debug("Expecting XRxxxxxxxx but got '{}'", line);
+            return;
+        }
+
+        final int dataType = Integer.parseInt(line.substring(2, 3), 16);
```

Comment:
I guess this is my habit, coming from C/C++ where I was used to write const (I know this is not the same ;) ) while programming embedded systems... Should I remove "final"s?